conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderAuxiliarySettersUtils.java

```diff
@@ -68,13 +70,19 @@ public static MethodSpec.Builder createOptionalSetterBuilder(
                 .addParameter(Parameters.nonnullParameter(
                         typeMapper.getClassName(type.getItemType()),
                         field.name,
-                        enriched.conjureDef().getSafety()));
+                        SafetyUtils.getMaybeExternalSafety(enriched.conjureDef())));
     }
 
     public static MethodSpec.Builder createItemSetterBuilder(
-            EnrichedField enriched, Type itemType, TypeMapper typeMapper, ClassName returnClass) {
+            EnrichedField enriched,
+            Type itemType,
+            TypeMapper typeMapper,
+            ClassName returnClass,
+            Optional<LogSafety> safety) {
         FieldSpec field = enriched.poetSpec();
-        return publicSetter(enriched, returnClass).addParameter(typeMapper.getClassName(itemType), field.name);
+        return publicSetter(enriched, returnClass)
```

Comment:
this is to support this https://github.com/palantir/conjure-java/pull/1944/files#r1103276431